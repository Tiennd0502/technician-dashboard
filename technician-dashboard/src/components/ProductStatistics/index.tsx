import { Flex, Heading, Image, Text, Button } from '@chakra-ui/react';
import { StatusLabel } from '..';
import { STATUS } from '@/interfaces';

const ProductStatistics = () => (
  <Flex
    flexDirection='column'
    justifyContent='space-between'
    borderRadius='md'
    p='7'
    borderWidth='1px'
    borderColor='border.primary'
    h='100%'
  >
    <Flex justifyContent='space-between' alignItems='center'>
      <Heading as='h3' variant='headingXl'>
        Total Products
      </Heading>
      <Text variant='text3Xl'>350</Text>
    </Flex>
    <Flex justifyContent='center' alignItems='center' width='100%'>
      <Image src='total-product.png' alt='Total product' maxW='min-content' />
    </Flex>
    <Flex justifyContent='space-between'>
      <Button variant='outline'>
        <StatusLabel value={STATUS.Activated} />
      </Button>
      <Button variant='outline'>
        <StatusLabel value={STATUS.Deactivated} />
      </Button>
    </Flex>
  </Flex>
);

export default ProductStatistics;
