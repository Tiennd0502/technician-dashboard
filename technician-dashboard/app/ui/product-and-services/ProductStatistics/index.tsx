import { Flex, Heading, Text, Button, Box } from '@chakra-ui/react';
import Image from 'next/image';

import { PRODUCT_STATISTICS_URL } from '@/lib/constants';
import { StatusLabel } from '@/ui/components';
import { STATUS } from '@/lib/interfaces';

interface ProductStatisticsProps {
  src?: string;
}

const ProductStatistics = ({ src = PRODUCT_STATISTICS_URL }: ProductStatisticsProps) => (
  <Flex
    flexDirection='column'
    justifyContent='space-between'
    borderRadius='md'
    p='7'
    borderWidth='1px'
    borderColor='border.primary'
    h='100%'
  >
    <Flex justifyContent='space-between' alignItems='center'>
      <Heading as='h3' variant='headingXl'>
        Total Products
      </Heading>
      <Text variant='text3Xl'>350</Text>
    </Flex>
    <Flex justifyContent='center' w='full' gap='2' mt='4'>
      <Box mt='auto' mb={{ base: '5', md: '0', lg: '7' }}>
        <Text>285</Text>
        <Text variant='textXs'>Activated</Text>
      </Box>
      <Box alignSelf='center' flexBasis={{ base: '200px', md: '320px' }}>
        <Image
          src={src}
          alt='Total product'
          style={{
            width: '100%',
            height: 'auto',
          }}
          width={200}
          height={180}
          blurDataURL={src}
          placeholder='blur'
        />
      </Box>
      <Box alignSelf='start' mb='4'>
        <Text>65</Text>
        <Text variant='textXs'>Deactivated</Text>
      </Box>
    </Flex>
    <Flex justifyContent='space-between' mt='4'>
      <Button variant='outline'>
        <StatusLabel value={STATUS.Activated} />
      </Button>
      <Button variant='outline'>
        <StatusLabel value={STATUS.Deactivated} />
      </Button>
    </Flex>
  </Flex>
);

export default ProductStatistics;
