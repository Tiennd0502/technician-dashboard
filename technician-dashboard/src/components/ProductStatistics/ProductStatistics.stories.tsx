import type { Meta, StoryObj } from '@storybook/react';

import ProductStatistics from '.';

const meta: Meta<typeof ProductStatistics> = {
  title: 'Components/ProductStatistics',
  component: ProductStatistics,
  tags: ['autodocs'],
};

export default meta;

type Story = StoryObj<typeof ProductStatistics>;

export const Default: Story = {
  args: {},
};
