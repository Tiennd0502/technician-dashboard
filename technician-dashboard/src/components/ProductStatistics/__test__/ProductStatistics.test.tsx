import { render } from '@testing-library/react';

import ProductStatistics from '..';

describe('ProductStatistics test cases', () => {
  test('should render correctly', () => {
    const { container } = render(<ProductStatistics />);
    expect(container).toMatchSnapshot();
  });
});
