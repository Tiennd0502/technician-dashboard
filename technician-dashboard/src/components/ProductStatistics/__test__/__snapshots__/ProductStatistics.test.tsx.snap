// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductStatistics test cases should render correctly 1`] = `
<div>
  <div
    class="css-kbu70o"
  >
    <div
      class="css-69i1ev"
    >
      <h3
        class="chakra-heading css-0"
      >
        Total Products
      </h3>
      <p
        class="chakra-text css-0"
      >
        350
      </p>
    </div>
    <div
      class="css-1pxgmwj"
    >
      <img
        alt="Total product"
        class="chakra-image css-1jonlwm"
        src="total-product.png"
      />
    </div>
    <div
      class="css-gg4vpm"
    >
      <button
        class="chakra-button css-4xx2wk"
        type="button"
      >
        <div
          class="css-70qvj9"
        >
          <div
            class="chakra-icon css-19b2uvg"
            focusable="false"
          />
          <p
            class="chakra-text css-0"
            data-testid="status-value"
          >
            Activated
          </p>
        </div>
      </button>
      <button
        class="chakra-button css-4xx2wk"
        type="button"
      >
        <div
          class="css-70qvj9"
        >
          <div
            class="chakra-icon css-crig9m"
            focusable="false"
          />
          <p
            class="chakra-text css-0"
            data-testid="status-value"
          >
            Deactivated
          </p>
        </div>
      </button>
    </div>
  </div>
</div>
`;
